serializers/src/main/java/io/pravega/schemaregistry/schemas/JSONSchema.java

```diff
@@ -0,0 +1,180 @@
+        this.schemaString = schemaString;
+        this.schemaInfo = new SchemaInfo(name, SerializationFormat.Json, getSchemaBytes(), ImmutableMap.of());
+        this.base = base;
+        this.derived = derived;
+        this.schema = schema;
+    }
+
+    private JSONSchema(SchemaInfo schemaInfo, JsonSchema schema, String schemaString, Class<T> derived) {
+        this.schemaString = schemaString;
+        this.schemaInfo = schemaInfo;
+        this.base = derived;
+        this.derived = derived;
+        this.schema = schema;
+    }
+
+    /**
+     * Method to create a typed JSONSchema for the given class. It extracts the json schema from the class.
+     * For POJOs the schema is extracted using jacksons {@link JsonSchemaGenerator}. 
+     *
+     * @param tClass Class whose object's schema is used.
+     * @param <T> Type of the Java class. 
+     * @return {@link JSONSchema} with generic type T that extracts and captures the json schema. 
+     */
+    public static <T> JSONSchema<T> of(Class<T> tClass) {
+        Preconditions.checkNotNull(tClass);
+        try {
+            JsonSchemaGenerator schemaGen = new JsonSchemaGenerator(OBJECT_MAPPER);
+            JsonSchema schema = schemaGen.generateSchema(tClass);
+            String schemaString = OBJECT_MAPPER.writeValueAsString(schema);
+
+            return new JSONSchema<>(schema, tClass.getName(), schemaString, tClass);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Unable to get json schema from the class", e);
+        }
+    }
+
+    /**
+     * Method to create a typed JSONSchema of type {@link Object} from the given schema. 
+     * This method can be used to pass Json schema string which can be used to represent primitive data types. 
+     *
+     * @param type type of object identified by {@link SchemaInfo#getType()}.
+     * @param schema Schema to use. 
+     * @return Returns an JSONSchema with {@link Object} type. 
+     */
+    public static JSONSchema<Object> of(String type, JsonSchema schema) {
+        Preconditions.checkNotNull(type);
+        Preconditions.checkNotNull(schema);
+        try {
+            String schemaString = OBJECT_MAPPER.writeValueAsString(schema);
+
+            return new JSONSchema<>(schema, type, schemaString, Object.class);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Unable to get json schema string from the JsonSchema object", e);
+        }
+    }
+
+    /**
+     * Method to create a typed JSONSchema of type {@link Object} from the given schema string. 
+     *
+     * @param type type of object identified by {@link SchemaInfo#getType()}.
+     * @param schemaString Schema string to use. 
+     * @return Returns an JSONSchema with {@link Object} type. 
+     */
+    public static JSONSchema<Object> of(String type, String schemaString) {
```

Comment:
though i understand that a type of Object is effectively same as erased type, I prefer to keep the type because it avoids unchecked warnings otherwise.

Reply:
Sorry github at my comment, the above actually reads:
```Should this be a JSONSchema<T>```
That would allow the caller to define the generic they want to avoid warnings.

Reply:
Actually, if users want to type it to T, they would use one of the Typed `of` methods. 
This method is when user is providing a schema string and not the typed object for creating the schema.

Reply:
i have changed this to take the type from the user.